File: java/Human.java
/*
Saya [Elsa Nabiilah] [2108805]
mengerjakan LATIHAN 2 dalam mata kuliah Desain Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/
/*
Saya [Elsa Nabiilah] [2108805]
mengerjakan LATIHAN 2 dalam mata kuliah Desain Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/
public class Human {
    private String NIK;
    private String nama;
    private String jk;

    public Human() {
        this.NIK = "";
        this.nama = "";
        this.jk = "";
    }

    public Human(String NIK, String nama, String jk) {
        this.NIK = NIK;
        this.nama = nama;
        this.jk = jk;
    }

    public String getnik() {
        return this.NIK;
    }

    public void setnik(String NIK) {
        this.NIK = NIK;
    }

    public String getnama() {
        return this.nama;
    }

    public void setnama(String nama) {
        this.nama = nama;
    }

    public String getjk() {
        return this.jk;
    }

    public void setjk(String jk) {
        this.NIK = jk;
    }

    void nik() {
        System.out.println("NIK           : " + this.NIK);
    }

    void nm() {
        System.out.println("Nama : " + this.nama);
    }

    void jenisk() {
        System.out.println("Jenis Kelamin : " + this.jk);
    }
}

File: cpp/Human.cpp
// package //cpp;
#include <bits/stdc++.h>
using namespace std;
/*
Saya [Elsa Nabiilah] [2108805]
mengerjakan LATIHAN 2 dalam mata kuliah Desain Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/
class Human
{
    // human menjadi class parent karena
    // meliputi keseluruhan objek class lain
    // artinya class lain memiliki objek Human

private:
    // deklarasi variabel
    string NIK;
    string nama;
    string jk;

public:
    // constructor
    Human()
    {
    }

    Human(string NIK, string nama, string jk)
    {
        this->NIK = NIK;
        this->nama = nama;
        this->jk = jk;
    }
    // getter setter method
    string getnik()
    {
        return this->NIK;
    }

    void setnik(string NIK)
    {
        this->NIK = NIK;
    }

    string getnama()
    {
        return this->nama;
    }

    void setnama(string nama)
    {
        this->nama = nama;
    }

    string getjk()
    {
        return this->jk;
    }

    void setjk(string jk)
    {
        this->jk = jk;
    }

    // function untuk menampilkan per atribut
    void printnik()
    {
        cout << "NIK           : " << this->getnik() << endl;
    }
    void printnama()
    {
        cout << "Nama          : " << this->getnama() << endl;
    }
    void printjk()
    {
        cout << "Jenis kelamin : " << this->getjk() << endl;
    }
    // destructor
    ~Human() {}
};
